File: DataGridView_Search_TextBox/DataGridView_Search_TextBox_VB/Form1.vb
Imports System.Data
Imports System.Data.SqlClient

Public Class Form1
    Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
        dataGridView1.DataSource = Me.PopulateDataGridView()
    End Sub

    Private Sub txtName_KeyUp(sender As System.Object, e As System.Windows.Forms.KeyEventArgs) Handles txtName.KeyUp
        dataGridView1.DataSource = Me.PopulateDataGridView()
    End Sub

    Private Function PopulateDataGridView() As DataTable
        Dim query As String = "SELECT CustomerID, ContactName, Country FROM Customers"
        query &= " WHERE ContactName LIKE '%' + @ContactName + '%'"
        query &= " OR @ContactName = ''"
        Dim constr As String = "Data Source=.\SQL2005;Initial Catalog=Northwind;User ID=sa;Password=pass@123"
        Using con As SqlConnection = New SqlConnection(constr)
            Using cmd As SqlCommand = New SqlCommand(query, con)
                cmd.Parameters.AddWithValue("@ContactName", txtName.Text.Trim())
                Using sda As SqlDataAdapter = New SqlDataAdapter(cmd)
                    Dim dt As DataTable = New DataTable()
                    sda.Fill(dt)
                    Return dt
                End Using
            End Using
        End Using
    End Function
End Class

File: DataGridView_Search_TextBox/DataGridView_Search_TextBox_CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Data;
using System.Data.SqlClient;

namespace DataGridView_CS
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            dataGridView1.DataSource = this.PopulateDataGridView();
        }

        private void txtName_KeyUp(object sender, KeyEventArgs e)
        {
            dataGridView1.DataSource = this.PopulateDataGridView();
        }

        private DataTable PopulateDataGridView()
        {
            string query = "SELECT CustomerID, ContactName, Country FROM Customers";
            query += " WHERE ContactName LIKE '%' + @ContactName + '%'";
            query += " OR @ContactName = ''";
            string constr = @"Data Source=.\SQL2005;Initial Catalog=Northwind;User ID=sa;Password=pass@123";
            using (SqlConnection con = new SqlConnection(constr))
            {
                using (SqlCommand cmd = new SqlCommand(query, con))
                {
                    cmd.Parameters.AddWithValue("@ContactName", txtName.Text.Trim());
                    using (SqlDataAdapter sda = new SqlDataAdapter(cmd))
                    {
                        DataTable dt = new DataTable();
                        sda.Fill(dt);
                        return dt;
                    }
                }
            }
        }
    }
}
